### PR TITLE
feat: verifyCloudProof function

### PR DESCRIPTION
**Pull request by 0xPenryn**
Adds the `verifyCloudProof` function, exported from `@worldcoin/idkit-core/backend` and `@worldcoin/idkit`. This function is intended to simplify the process of verifying a proof via the Developer Portal API, and will use the new `/api/v2/verify/[app_id]` endpoint once available.

### CODE REVIEW

File: examples/with-next/pages/api/verify.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import { verifyCloudProof, IVerifyResponse } from '@worldcoin/idkit'
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+	if (req.method === 'POST') {
+		const { proof, app_id, action, signal } = req.body
+		const response = (await verifyCloudProof(proof, app_id, action, signal)) as IVerifyResponse
+		res.status(response.success ? 200 : 400).json(response)
+	} else {
+		res.status(405).json({ message: 'Method Not Allowed' })
+	}
```
Comment (m1guelpf) [inline examples/with-next/pages/api/verify.ts:11]:
```suggestion
	if (req.method != 'POST') return res.status(405).json({ message: 'Method Not Allowed' })
	
	const { proof, app_id, action, signal } = req.body
	const response = (await verifyCloudProof(proof, app_id, action, signal)) as IVerifyResponse
	res.status(response.success ? 200 : 400).json(response)
```

File: packages/core/src/lib/backend.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+export async function verifyCloudProof(
+	proof: ISuccessResult,
+	app_id: `app_${string}`,
+	action: string,
+	signal?: string
+): Promise<IVerifyResponse> {
+	if (!isNode && !isDeno) {
+		throw new Error('verifyCloudProof can only be used in the backend.')
+	}
```
Comment (m1guelpf) [inline packages/core/src/lib/backend.ts:20]: Can we just do `isBrowser` instead (from the same package)? seems more resilient

File: packages/core/src/lib/backend.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+	if (!endpoint) {
+		endpoint = new URL('https://developer.worldcoin.org/api/v2/verify')
+	}
+
+	if (typeof endpoint === 'string') {
+		endpoint = new URL(endpoint)
+	}
+
+	const response = await fetch(endpoint.href + '/' + app_id, {
```
Comment (andy-t-wang) [inline packages/core/src/lib/backend.ts:31]: What's the point of doing this !endpoint. If it's to let people pass in their own we shouldn't append anything to it. We can append app_id into our own oversion

File: examples/with-next/pages/index.tsx [hunk 4]
```diff
@@ -1,23 +1,49 @@
-import { IDKitWidget, VerificationLevel } from '@worldcoin/idkit'
 import { GetServerSideProps, InferGetServerSidePropsType } from 'next'
+import { IDKitWidget, ISuccessResult, IVerifyResponse, VerificationLevel } from '@worldcoin/idkit'
 
 export const getServerSideProps = (async context => {
 	return {
 		props: {
 			app_id:
 				(context.query.app_id?.toString() as `app_${string}`) || 'app_staging_45068dca85829d2fd90e2dd6f0bff997',
+			action: (context.query.action?.toString() as string) || 'test-action',
```
Comment (andy-t-wang) [inline examples/with-next/pages/index.tsx:9]: The example should use the new v2 verify then?

File: packages/core/src/lib/backend.ts (hunk 3 above)
Comment (0xPenryn) [reply packages/core/src/lib/backend.ts]: mainly it's convenience for testing this against the staging env -- but i agree with your point that we shouldn't modify an endpoint passed in, fixed now

**Review by andy-t-wang — APPROVED** [review]

**Comment by 0xPenryn** [comment]
wait to merge until tested against /api/v2/verify endpoint in staging

**Comment by Tobiasguilleron** [comment]
Id2.0

**Comment by Tobiasguilleron** [comment]
@wld-id-sync quiero ser parte de la app mundial

**Review by gatleas17 — COMMENTED** [review]
Hi

**Pull request merged**